js-classes/inheritance-es5.js

```diff
@@ -0,0 +1,93 @@
+function Car(name,model,year,color,maxSpeed,fuelCapacity=60,fuelConsumption=10)
+    {
+     this.name=name;
+     this.model=model;
+     this.year=year;
+     this.color=color;
+     this.maxSpeed=maxSpeed;
+     this.fuelCapacity=fuelCapacity;
+     this.fuelConsumption=fuelConsumption;   
+    }
+Car.prototype.getFullName=function()
+    {
+        return `${this.name}+${this.model}`;
+    }
+Car.prototype.getAge=function()
+{
+    let currentYear=2020;
+    return currentYear-`${this.year}`;
+}
+Car.prototype.changeColor=function(color)
+{
+    if(this.color==color)
+    {
+        return 'It is the same color: ' + this.color;
+    }
+    else
+    {
+        return 'The color was changed: ' + (this.color = color);
+    }
+}
+Car.prototype.calculateWay=function(kilometers,fuel)
+{
+    if(fuel<10)
+    {
+        alert('Better refuel')
+    }
+    let speed = this.maxSpeed / 2;
+    let timeToReachDestination = Math.ceil(kilometers / speed * 60);
+    let needToFuel = kilometers / this.fuelConsumption - fuel;
+    let timesToRefuel = Math.ceil(needToFuel / this.fuelCapacity);
+
+    console.log(`You need ${timeToReachDestination} minutes at the speed ${speed} to reach the destination.`);
+
+    if (needToFuel <= 0) {
+        return 'You have enough fuel to reach the destination';
+    }
+    else {
+        return `You need ${needToFuel} liters of the fuel to reach the destination. You need ${timesToRefuel} time to refuel.`;
+    }
+}
+function BMW(name,model,year,color,maxSpeed,fuelCapacity=60,fuelConsumption=10,garanty=2)
+{
+    Car.call(this,name,model,year,color,maxSpeed,fuelCapacity,fuelConsumption);
+    this.garanty=garanty;
+}
+function Mercedes(name,model,year,color,maxSpeed,fuelCapacity=60,fuelConsumption=10,colorLight)
+{
+    Car.call(this,name,model,year,color,maxSpeed,fuelCapacity,fuelConsumption);
+    this.colorLight=colorLight;
+}
+function Kia(name,model,year,color,maxSpeed,fuelCapacity=60,fuelConsumption=10,automaticParking)
+{
+    Car.call(this,name,model,year,color,maxSpeed,fuelCapacity,fuelConsumption);
+    this.automaticParking=automaticParking;
+}
+
+BMW.prototype=Object.create(Car.prototype);
```

Comment:
The constructor should be saved:
```suggestion
BMW.prototype=Object.create(Car.prototype);
BMW.prototype.constructor = BMW
```